src/pybind/ceph_volume_client.py

```diff
@@ -1442,6 +1451,34 @@ def get_object(self, pool_name, object_name):
             ioctx.close()
         return bytes_read
 
+    def update_object(self, pool_name, object_name, data_modifier):
```

Comment:
This API works but I'd prefer to see a push-style API. How about also add `def put_object_versioned` which this method also calls.

Reply:
@batrick did you see my earlier comment,
"
If you already know the data to write to object, then you can just use put_object, which is atomic, right?

Here you want to make changes to the data fetched from object, and put the result back in the object. Instead of 'data' parameter, you'd want the processing function, its *args, and **kwargs, that'd make the desired changes to the fetched data, to be passed in as arguments to 'update_object'.

The other approach would be to do the object update (get data, process fetched data, put result) at a higher level (e.g. the manila cephfs driver). In this library provide interfaces to

    fetch object data along with its version: get_object_and version(pool_name, object_name) that returns (obj_data, obj_version)
    do compound write op to assert object version and then write to object using python bindings for 'rados_write_op_assert_version' followed by 'rados_write_op_write_full' [1]: put_object_with_assert_version(pool_name, object_name, data, version)
    In the manila driver you can do something like,

while True:
      obj_data, obj_version = get_object_and_version(pool_name, object_name)
      result_data = foo(obj_data)
      try:
           put_object_with_assert_version(pool_name, object_name, result_data, obj_version)
      except versionmismatch:
            continue
      break

@batrick what do you think?

[1] See, https://github.com/ceph/ceph/blob/master/src/pybind/rados/rados.pyx#L1970
For usage example see, https://github.com/ceph/ceph/blob/master/src/test/pybind/test_rados.py#L509
"

Reply:
@batrick sugggests, Rishabh
-  we don't need a separate **get_object_and version()**method just modify the existing **get_object()** that also returns object version, and bump up the CephFSVolumeClient version

- add  **put_object_versioned()** , a more intuitive name for the method **put_object_with_assert_version()** that  I described above.

Reply:
What if additional parameters besides data fetched from get_object() need to be passed to the function  **data_modifier()**?

I think it's better if you just expose two methods in this  library
- **get_object()** that returns object and its version
- ** put_object_versioned()** that does a compound writeop (checks version and then writes to object)

These two methods should be good enough for the application using this library to implement atomic update_object()

Reply:
> we don't need a separate get_object_and version() method just modify the existing get_object() such that it also returns the object version, and bump up the CephFSVolumeClient version

I misunderstood batrick's suggestion. Thanks, rishabh.  So as you said,
implement, get_object_and_version(), and modify get_object() to
return get_object_and_version()['data']